[BUG] Map.clear报错java.lang.NoClassDefFoundError: Failed resolution of:

### 安卓版本
> 11

### 阅读Legdao版本（我的-关于-版本,如3.20.112220）
> 3.22.0530008

### 问题描述（简要描述发生的问题）
> js Map.clear()报错 ，正式版3.22.052723没问题

### 复现步骤（详细描述导致问题产生的操作步骤，如果能稳定复现）
> 添加下面的书源 ,然后调试


### 使用书源（填写URL或者JSON）

```json
{
  "bookSourceComment": "",
  "bookSourceName": "Map复现",
  "bookSourceType": 0,
  "bookSourceUrl": "http://baidu.cn",
  "customOrder": 0,
  "enabled": true,
  "enabledCookieJar": true,
  "enabledExplore": true,
  "lastUpdateTime": 1653881774793,
  "respondTime": 180000,
  "ruleBookInfo": {
    "init": ""
  },
  "ruleContent": {},
  "ruleExplore": {
    "bookList": ""
  },
  "ruleSearch": {
    "bookList": ""
  },
  "ruleToc": {
    "chapterList": ""
  },
  "searchUrl": "<js>\nlet a = new Map();\na.clear()\n</js>",
  "weight": 0
}
```

<h3>日志提交（我的-关于-崩溃日志，书架-右上角-日志）</h3>
<details>
 <summary>点击展开</summary>
 <pre><code>

java.lang.NoClassDefFoundError: Failed resolution of: Lj$/util/Iterator$-EL;
	at org.mozilla.javascript.Hashtable.clear(Hashtable.java:201)
	at org.mozilla.javascript.NativeMap.js_clear(NativeMap.java:139)
	at org.mozilla.javascript.NativeMap.execIdCall(NativeMap.java:72)
	at org.mozilla.javascript.IdFunctionObject.call(IdFunctionObject.java:100)
	at org.mozilla.javascript.Interpreter.interpretLoop(Interpreter.java:1692)
	at org.mozilla.javascript.Interpreter.interpret(Interpreter.java:1013)
	at org.mozilla.javascript.InterpretedFunction.call(InterpretedFunction.java:109)
	at org.mozilla.javascript.ContextFactory.doTopCall(ContextFactory.java:412)
	at com.script.javascript.RhinoScriptEngine$1.superDoTopCall(RhinoScriptEngine.java:92)
	at com.script.javascript.RhinoScriptEngine$1.doTopCall(RhinoScriptEngine.java:85)
	at org.mozilla.javascript.ScriptRuntime.doTopCall(ScriptRuntime.java:3651)
	at org.mozilla.javascript.InterpretedFunction.exec(InterpretedFunction.java:121)
	at org.mozilla.javascript.Context.evaluateReader(Context.java:1265)
	at com.script.javascript.RhinoScriptEngine.eval(RhinoScriptEngine.java:161)
	at com.script.javascript.RhinoScriptEngine.eval(RhinoScriptEngine.java:187)
	at com.script.AbstractScriptEngine.eval(AbstractScriptEngine.java:79)
	at io.legado.app.model.analyzeRule.AnalyzeUrl.evalJS(AnalyzeUrl.kt:252)
	at io.legado.app.model.analyzeRule.AnalyzeUrl.analyzeJs(AnalyzeUrl.kt:117)
	at io.legado.app.model.analyzeRule.AnalyzeUrl.initUrl(AnalyzeUrl.kt:95)
	at io.legado.app.model.analyzeRule.AnalyzeUrl.<init>(AnalyzeUrl.kt:85)
	at io.legado.app.model.analyzeRule.AnalyzeUrl.<init>(AnalyzeUrl.kt:36)
	at io.legado.app.model.webBook.WebBook.searchBookAwait(WebBook.kt:44)
	at io.legado.app.model.webBook.WebBook$searchBook$1.invokeSuspend(WebBook.kt:32)
	at io.legado.app.model.webBook.WebBook$searchBook$1.invoke(Unknown Source:8)
	at io.legado.app.model.webBook.WebBook$searchBook$1.invoke(Unknown Source:4)
	at io.legado.app.help.coroutine.Coroutine$executeBlock$2.invokeSuspend(Coroutine.kt:200)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.internal.LimitedDispatcher.run(LimitedDispatcher.kt:42)
	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:95)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:570)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:749)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:677)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:664)
Caused by: java.lang.ClassNotFoundException: j$.util.Iterator$-EL
	... 34 more

<pre></code>
</details>

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -253,7 +253,7 @@ dependencies {
     implementation('com.github.liuyueyi.quick-chinese-transfer:quick-transfer-core:0.2.4')
 
     //加解密类库,有些书源使用
-    implementation('cn.hutool:hutool-crypto:5.8.2')
+    implementation('cn.hutool:hutool-crypto:5.8.1')
 
     //firebase, 崩溃统计和性能统计, 会导致共存版本崩溃
     //implementation platform('com.google.firebase:firebase-bom:30.0.1')

--- a/app/src/main/res/values-es-rES/strings.xml
+++ b/app/src/main/res/values-es-rES/strings.xml
@@ -991,7 +991,6 @@
     <string name="add_remote_book">WebDavBook</string>
     <string name="bitmap_cache_size_summary">Current cache max size %1$s MB</string>
     <string name="bitmap_cache_size">bitmap cache size</string>
-    <string name="restart_to_apply_changes">Restart to apply changes</string>
     <string name="export_pics_file">Export Picture Files</string>
     <!-- string end -->
 

--- a/app/src/main/res/values-ja-rJP/strings.xml
+++ b/app/src/main/res/values-ja-rJP/strings.xml
@@ -994,7 +994,6 @@
     <string name="add_remote_book">WebDavBook</string>
     <string name="bitmap_cache_size_summary">Current cache max size %1$s MB</string>
     <string name="bitmap_cache_size">bitmap cache size</string>
-    <string name="restart_to_apply_changes">Restart to apply changes</string>
     <string name="export_pics_file">Export Picture Files</string>
     <!-- string end -->
 

--- a/app/src/main/res/values-pt-rBR/strings.xml
+++ b/app/src/main/res/values-pt-rBR/strings.xml
@@ -994,7 +994,6 @@
     <string name="add_remote_book">WebDavBook</string>
     <string name="bitmap_cache_size_summary">Current cache max size %1$s MB</string>
     <string name="bitmap_cache_size">bitmap cache size</string>
-    <string name="restart_to_apply_changes">Restart to apply changes</string>
     <string name="export_pics_file">Export Picture Files</string>
     <!-- string end -->
 

--- a/app/src/main/res/values-zh-rHK/strings.xml
+++ b/app/src/main/res/values-zh-rHK/strings.xml
@@ -991,7 +991,6 @@
     <string name="add_remote_book">WebDav书籍</string>
     <string name="bitmap_cache_size_summary">当前最大缓存 %1$s MB</string>
     <string name="bitmap_cache_size">图片绘制缓存</string>
-    <string name="restart_to_apply_changes">重启应用更改</string>
     <string name="export_pics_file">TXT导出图片</string>
     <!-- string end -->
 

--- a/app/src/main/res/values-zh-rTW/strings.xml
+++ b/app/src/main/res/values-zh-rTW/strings.xml
@@ -993,7 +993,6 @@
     <string name="add_remote_book">WebDav书籍</string>
     <string name="bitmap_cache_size_summary">当前最大缓存 %1$s MB</string>
     <string name="bitmap_cache_size">图片绘制缓存</string>
-    <string name="restart_to_apply_changes">重启应用更改</string>
     <string name="export_pics_file">TXT导出图片</string>
     <!-- string end -->
 

--- a/app/src/main/res/values-zh/strings.xml
+++ b/app/src/main/res/values-zh/strings.xml
@@ -993,7 +993,6 @@
     <string name="add_remote_book">WebDav书籍</string>
     <string name="bitmap_cache_size_summary">当前最大缓存 %1$s MB</string>
     <string name="bitmap_cache_size">图片绘制缓存</string>
-    <string name="restart_to_apply_changes">重启应用更改</string>
     <string name="export_pics_file">TXT导出图片</string>
     <!-- string end -->
 

--- a/app/src/main/res/values/strings.xml
+++ b/app/src/main/res/values/strings.xml
@@ -994,7 +994,6 @@
     <string name="add_remote_book">WebDavBook</string>
     <string name="bitmap_cache_size_summary">Current cache max size %1$s MB</string>
     <string name="bitmap_cache_size">bitmap cache size</string>
-    <string name="restart_to_apply_changes">Restart to apply changes</string>
     <string name="export_pics_file">Export Picture Files</string>
     <!-- string end -->
 

